### DISCUSSION TITLE
SQLITE "Change Filesize Bigint" migration fails

### DISCUSSION THREAD

**Discussion by Thomas-lhuillier** [directus/directus#7391] [Archive]
Hello folks,

I'm trying to migrate my SQLITE database but it fails at `Applying Change Filesize Bigint...`. Which I  believe is part of `9.0.0-rc.83`.

```
✨ Applying Change Filesize Bigint...
[Error: DROP TABLE "directus_files" - SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] {
  errno: 19,
  code: 'SQLITE_CONSTRAINT'
}
```

Any insights regarding that error?

**Comment by rijkvanzanten** [comment]
I _think_ that might be caused by this known issue in `knex`: https://github.com/knex/knex/issues/4524

**Reply by Thomas-lhuillier** [reply]
Sounds right 🙇

**Reply by rijkvanzanten** [reply]
@nickrum I know it has been a while since we did the sqlite improvements on `knex`, but is this something that's something we could address? It seems like the issue went stale

**Reply by nickrum** [reply]
@rijkvanzanten Absolutely! I wanted to move all of the DDL functions to the parser based approach anyway. That should fix the issue in the process.
